### PR TITLE
Advanced multipart request example + minor fixes

### LINKED ISSUE #203: Add FormData example
After https://github.com/loadimpact/k6/pull/1800 and https://github.com/loadimpact/jslib.k6.io/pull/21 are merged (i.e. k6 v0.31.0), we should add an example that demonstrates how to build advanced `multipart/form-data` request bodies with them. As evidenced by https://github.com/loadimpact/k6/issues/747, https://github.com/loadimpact/k6/issues/843, https://github.com/loadimpact/k6/issues/1375, https://github.com/loadimpact/k6/issues/1382, https://github.com/loadimpact/k6/issues/1571, https://github.com/loadimpact/k6/issues/1784, the current k6 HTTP API is quite restricted and can't cover a lot of corner cases... :disappointed: We'll close these issues (https://github.com/loadimpact/k6/pull/1800#issuecomment-768157584), so we probably should also add warning messages in https://k6.io/docs/javascript-api/k6-http/file-data-filename-contenttype and https://k6.io/docs/javascript-api/k6-http/filedata to use the JS polyfill for the corner cases the Go API can't cover.

At some future point, we'll try to fix all of these deficiencies with a new and better k6 JS API for HTTP requests, but until then this polyfill will likely be sufficient for 99% of users, with almost no downsides. So, we should document it properly.

### PR DESCRIPTION
**Pull request by imiric**
Closes #203

We should merge https://github.com/loadimpact/jslib.k6.io/pull/21 and deploy first, otherwise the `FormData` links won't work.

### CODE REVIEW

File: src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md
Comment (MStoykov) [inline src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md:96]: nit: This makes it sound like JS objects don't have an order to their keys ... but they do. I am pretty sure that even in goja that works. The problem is that they are converted to golang maps, which don't, but I am not certain this is made clear from the wording here. I tried fixing it ... but with no success :(

File: src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md
Comment (MStoykov) [inline src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md:117]: nit: this can also be `const` as `const` is only about the identifier - it prevents the identifier to point/identify another object, after initialization.

**Review by MStoykov — APPROVED** [review]
LGTM. 

I do think we should probably add a warning to users that using the Object body has the problems you've outlined everywhere. But:
1. this can be done in another PR
2. it will probably be better done after we have tested the FormData somewhat more :) and have gotten at least some reports that it works great for users as well :)

**Review by ppcano — APPROVED** [review]

File: src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md
Comment (imiric) [reply src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md:96]: AFAIK key order was unspecified in <=ES5, and browsers did their own implementation. Even in >=ES6 [the situation is messy](https://stackoverflow.com/a/30919039/96213), and integer keys are ordered numerically, while others are ordered by insertion.

I'm not sure how goja handles this internally, but for our purposes (and I'd argue in general for JS) it's clearer to not rely on objects for order, and use something like `Map` which explicitly supports it.

I thought the phrasing here is clear that it's an issue when they're converted to Golang maps and didn't want to go into any more detail. So suggestions for how to change it is welcome.

File: src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md
Comment (imiric) [reply src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md:117]: Thanks, I was aware of that, but wasn't thorough enough with the usage. Will change it.

File: src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md
Comment (MStoykov) [reply src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md:96]: It just has the property names as a list  ;) 
https://github.com/dop251/goja/blob/f5884268f0c0a0eb94423dcf7c2eda14907375e6/object.go#L191-L199

as I said, this is a nitpick. Unless somebody else comes up with better wording, as I can't myself as well, it is fine to just merge it as is.

File: src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md
Comment (imiric) [reply src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md]: Resolved in 397babd.

File: src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md
Comment (simskij) [inline src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md:105]: Please use the `CodeGroup` component instead:

```
<CodeGroup labels={["Nice code!", "This one is better", "Oh my.."]} lineNumbers={[true, true, true]}>
   ...
</CodeGroup
```

**Review by simskij — CHANGES_REQUESTED** [review]
One minor detail. After that, feel free to merge without a second review. 👍🏼

File: src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md
Comment (imiric) [reply src/data/markdown/docs/05 Examples/01 Examples/09 data-uploads.md]: Thanks, I think copied this from an old part of the docs. Will keep it in mind for the future.

**Pull request merged**
